assopy/middleware.py

```diff
@@ -15,33 +15,33 @@ def dump_exc_plus(exclude=None):
 
     traceback.print_exc(file=s)
 
-    print >>s, "Locals by frame, innermost last"
+    print("Locals by frame, innermost last", file=s)
     if exclude:
-        print >>s, "  excluding '%s'" % "','".join(exclude)
+        print("  excluding '%s'" % "','".join(exclude), file=s)
 
     tb = sys.exc_info()[2]
     while tb:
         frame = tb.tb_frame
         tb = tb.tb_next
 
-        print >>s
-        print >>s, "Frame %s in %s at line %s" % (
+        print(file=s)
+        print("Frame %s in %s at line %s" % (
             frame.f_code.co_name,
             frame.f_code.co_filename,
-            frame.f_lineno)
+            frame.f_lineno), file=s)
 
-        for key, value in frame.f_locals.items():
+        for key, value in list(frame.f_locals.items()):
```

Comment:
Unnecessary `list`.

Reply:
For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;